libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/LayoutSyncManager.java

```diff
@@ -58,11 +59,10 @@
     private static final String SOUP_NAME = "sfdcLayouts";
     private static final String FEATURE_LAYOUT_SYNC = "LY";
     private static final String QUERY = "SELECT {" + SOUP_NAME + ":_soup} FROM {" + SOUP_NAME +
-            "} WHERE {" + SOUP_NAME + ":sobjectType} = '%s' AND {" + SOUP_NAME + ":layoutType} = '%s'";
+            "} WHERE {" + SOUP_NAME + ":" + Constants.ID + "} = '%s-%s'";
     private static final String TAG = "LayoutSyncManager";
     private static final IndexSpec[] INDEX_SPECS = new IndexSpec[] {
-        new IndexSpec("sobjectType", SmartStore.Type.json1),
```

Comment:
Should we keep those indexes if someone wanted to do lookup by object types for instance?
(With json1, it doesn't add columns to the table - so it has little impact on storage)

Reply:
We could, but there's little value in that though since the API callback already has `objectType` and the API calls have `objectType` too. I figured I'd try to keep the stored payload as close to the original as possible.

Reply:
See [here](https://github.com/bhariharan/SalesforceMobileSDK-Android/blob/dev/libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/LayoutSyncDownTarget.java#L53) - `objectType` is an instance variable in the sync down target so it's always available anyway.